specs/APIReview_UAString.md

```diff
@@ -0,0 +1,92 @@
+	For more info about the ‘—user - agent’ flag visit: https://peter.sh/experiments/chromium-command-line-switches/#user-agent.
+
+However, there are a couple limitations to this workaround-- you cannot modify a command line switch at runtime, nor can you change the user agent per webview. In this document we describe the new API. We'd appreciate your feedback.
+
+# Description
+
+The User Agent (UA) is a client-side piece of information that the browser/webcontrol sends to the server or website a user visits. It contains information about user’s system and is modifiable by the user.
+
+The User Agent String API allows developers to modify WebView's user agent string via Chrome Developer Protocol (CDP). The user agent can be changed based on different events such as navigation to a specific website. 
+
+The Settings component will change the UA per WebView2 via Chrome Developer Protocol command (CDP). A key scenario is to allow end developers to get the current user agent from the webview and modify it based on some sort of event, such as navigating to a specific website and setting the user agent to emulate a different browser version.
+
+# Examples
+
+The following code snippet demonstrates how the environment APIs can be used
+:
+
+## Win32 C++
+    
+```cpp 
+m_webView->add_NavigationStarting(
+    Callback<ICoreWebView2NavigationStartingEventHandler>(
+        [this](ICoreWebView2 *sender,
+                ICoreWebView2NavigationStartingEventArgs *args) -> HRESULT {
+            static const PCWSTR url_compare_example = L"foo.org";
+            wil::unique_bstr domain = GetDomainOfUri(uri.get());
+            const wchar_t *domains = domain.get();
+            if (wcscmp(url_compare_example, domains) == 0) {
+                // Upon navigation to a specified url 
+                // Change the user agent to emulate a mobile device 
+                wil::com_ptr<ICoreWebView2Settings> settings;
+                CHECK_FAILURE(m_webView->get_Settings(&m_settings));
+                LPCWSTR mobile_ua =
+                    "Mozilla/5.0 (Linux; Android 8.0.0; SM-G960F Build/R16NW) "
+                    "AppleWebKit/537.36 (KHTML, like Gecko) "
+                    "Chrome/62.0.3202.84 Mobile Safari/537.36";
+                CHECK_FAILURE(settings->put_UserAgent(mobile_ua));
+                LPCWSTR received_ua;
+                CHECK_FAILURE(settings->get_UserAgent(&received_ua));
+                EXPECT_STREQ(received_ua, mobile_ua);
+            }
+            return S_OK;
+        })
+        .Get(),
+    &m_navigationStartingToken);
+``` 
+
+## .NET and WinRT
+
+```c #
+private void SetUserAgent(CoreWebView2 sender, CoreWebView2UserAgentArgs e) {
+    var settings = webView2Control.CoreWebView2.Settings;
+    settings.UserAgent = "Mozilla/5.0 (Linux; Android 8.0.0; SM-G960F Build/R16NW) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/62.0.3202.84 Mobile Safari/537.36";
```

Comment:
We probably don't want to embed an actual UA here. How about we use a static constant string that we exclude the defintion of.

Reply:
Should that be passes in as an arg or in the function body: 
static const std::string updatedUA;
settings.UserAgent = updatedUA;

Reply:
Sorry didn't see this question. I was imagining a UA string constant defined elsewhere not shown or a method that returns it not shown, or a parameter would also work.